test/nursery/autogluon_for_single_ts.py

```diff
@@ -0,0 +1,120 @@
+from gluonts.model.predictor import Predictor
+# Standard library imports
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Iterator,
+    Iterable, Dict,
+)
+
+# Third-party imports
+import numpy as np
+from pandas.tseries.holiday import USFederalHolidayCalendar as calendar
+import pandas as pd
+from autogluon import TabularPrediction as task
+
+# First-party imports
+from gluonts.dataset.common import DataEntry, Dataset, ListDataset
+from gluonts.dataset.util import to_pandas
+from gluonts.model.forecast import Forecast, SampleForecast
+from gluonts.model.predictor import Localizer
+from gluonts.model.estimator import Estimator
+
+if TYPE_CHECKING:  # avoid circular import
+    from gluonts.model.estimator import Estimator  # noqa
+
+OutputTransform = Callable[[DataEntry, np.ndarray], np.ndarray]
+
+
+class TabularPredictor(Predictor):
+    def __init__(self,
+                 ag_model,
+                 freq: str,
+                 prediction_length: int,
+                 ) -> None:
+        self.ag_model = ag_model  # task?
+        self.freq = freq
+        self.prediction_length = prediction_length
+
+    def predict(self, dataset: Iterable[Dict], lag=0, model=None, as_pandas=False) -> Iterator[SampleForecast]:
+        def get_prediction_dataframe(series, lag=0, model=None, as_pandas=False):
```

Comment:
make sure that you annotate all input arguments and return types, wherever possible